partner_student_course/models/partner.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# (c) 2016 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, fields, api
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.multi
+    def _compute_student_course(self):
+        course_obj = self.env['partner.student.course']
+        for partner in self:
```

Comment:
`for partner in self.filtered(lambda x: x .birthdate_date):` con esto te ahorras el iterar todos los registros y el if siguiente